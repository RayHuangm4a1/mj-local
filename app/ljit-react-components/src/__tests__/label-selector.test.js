import React from 'react';
import { shallow, mount, render, } from 'enzyme';
import LabelSelector from '../components/label-selector';

const { OrientationTypeEnums, } = LabelSelector;

describe('Label Selector', () => {
	it('handle default props', () => {
		const {
			onClickItem,
			isSelectedShowBorder,
			orientation,
		} = LabelSelector.defaultProps;

		expect(orientation).toEqual(OrientationTypeEnums.HORIZONTAL);
		expect(onClickItem).toBeDefined();
		expect(onClickItem).toBeInstanceOf(Function);
		expect(isSelectedShowBorder).toBe(true);
	});

	it('should renders correctly', () => {
		const wrapper = render(
			<LabelSelector
				label="mock-label"
				items={[
					{
						id: 0,
						name: 'mock-name',
					},
				]}
				selectedId={0}
				onClickItem={() => {}}
				isSelectedShowBorder={true}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-label-selector', () => {
		const wrapper = shallow(
			<LabelSelector
				label="mock-label"
				items={[
					{
						id: 0,
						name: 'mock-name',
					},
				]}
				selectedId={0}
				onClickItem={() => {}}
				isSelectedShowBorder={true}
			/>
		);

		expect(wrapper.hasClass('ljit-label-selector')).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const label = 'mock-label';
		const items = [
			{
				id: 0,
				name: 'mock-name',
			},
		];
		const selectedId = 0;
		const onClickItem = () => {};
		const isSelectedShowBorder = true;
		const orientation = OrientationTypeEnums.VERTICAL;

		const wrapper = mount(
			<LabelSelector
				label={label}
				items={items}
				selectedId={selectedId}
				onClickItem={onClickItem}
				isSelectedShowBorder={isSelectedShowBorder}
				orientation={orientation}
			/>
		);

		expect(wrapper.props().label).toBe(label);
		expect(wrapper.props().items).toBe(items);
		expect(wrapper.props().selectedId).toBe(selectedId);
		expect(wrapper.props().orientation).toBe(orientation);
		expect(wrapper.props().onClickItem).toBe(onClickItem);
		expect(wrapper.props().isSelectedShowBorder).toEqual(isSelectedShowBorder);
	});

	it('should handle onClickItem', () => {
		const onClickItem = jest.fn();
		const item = {
			id: 0,
			name: 'mock-name',
		};
		const wrapper = mount(<LabelSelector
			label="mock-label"
			items={[ item, ]}
			selectedId={0}
			onClickItem={onClickItem}
			isSelectedShowBorder={true}
		/>);

		wrapper.find('.ljit-clickable-tag').simulate('click');
		expect(onClickItem).toHaveBeenCalledTimes(1);
		expect(onClickItem).toHaveBeenCalled();
		expect(onClickItem).toHaveBeenCalledWith(item);
	});
});

describe('OrientationTypeEnums ', () => {
	it('should contains HORIZONTAL property', () => {
		const typeName = 'horizontal';
		const formatType = 'HORIZONTAL';

		expect(OrientationTypeEnums[formatType]).toEqual(typeName);
	});

	it('should contains VERTICAL property', () => {
		const typeName = 'vertical';
		const formatType = 'VERTICAL';

		expect(OrientationTypeEnums[formatType]).toEqual(typeName);
	});
});
