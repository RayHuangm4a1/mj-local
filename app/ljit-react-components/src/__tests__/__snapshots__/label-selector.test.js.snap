// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Selector should renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ant-row ant-form-item ljit-label ljit-label-selector",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ant-col ant-form-item-label",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "",
              "title": "mock-label",
            },
            "children": Array [
              Object {
                "data": "mock-label",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "label",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "title": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "title": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "ant-col ant-form-item-control-wrapper",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "ant-form-item-control",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "ant-form-item-children",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "ljit-clickable-tag ljit-label-selector__item ljit-label-selector__item--active ljit-label-selector__item--active-border",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "ant-tag ljit-tag ljit-tag-color--black ljit-tag-shape--rectangle",
                            "style": "background-color:;color:",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {
                                "class": "ljit-tag__text",
                              },
                              "children": Array [
                                Object {
                                  "data": "mock-name",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "text",
                                },
                              ],
                              "name": "span",
                              "namespace": "http://www.w3.org/1999/xhtml",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                              "x-attribsNamespace": Object {
                                "class": undefined,
                              },
                              "x-attribsPrefix": Object {
                                "class": undefined,
                              },
                            },
                          ],
                          "name": "div",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                            "style": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                            "style": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "ant-col ant-form-item-control-wrapper",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "ant-form-item-control",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "ant-form-item-children",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "ljit-clickable-tag ljit-label-selector__item ljit-label-selector__item--active ljit-label-selector__item--active-border",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "ant-tag ljit-tag ljit-tag-color--black ljit-tag-shape--rectangle",
                          "style": "background-color:;color:",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "class": "ljit-tag__text",
                            },
                            "children": Array [
                              Object {
                                "data": "mock-name",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "text",
                              },
                            ],
                            "name": "span",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "style": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "style": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "ant-col ant-form-item-label",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "",
                "title": "mock-label",
              },
              "children": Array [
                Object {
                  "data": "mock-label",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "label",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "title": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "title": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
